server/src/main/java/org/candlepin/resource/ActivationKeyResource.java

```diff
@@ -279,9 +302,28 @@ public ActivationKey removeProductIdFromKey(
      */
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public List<ActivationKey> findActivationKey() {
-        List<ActivationKey> keyList = activationKeyCurator.listAll();
-        return keyList;
+    public Response findActivationKey() {
+        // TODO: Replace this with use of a cursor/iterator
+        final List<ActivationKey> keyList = activationKeyCurator.listAll();
+        final ObjectMapper mapper = new JsonProvider(true)
+            .locateMapper(Object.class, MediaType.APPLICATION_JSON_TYPE);
+
+        StreamingOutput output = new StreamingOutput() {
+            @Override
+            public void write(OutputStream stream) throws IOException, WebApplicationException {
+                JsonGenerator generator = mapper.getJsonFactory().createGenerator(stream);
+                generator.writeStartArray();
+
+                for (ActivationKey key : keyList) {
+                    mapper.writeValue(generator, new ActivationKeyData(key));
```

Comment:
Do you need to keep creating new ActivationKeyData objects? Why not re-use one and reload it using the fromModel() method?